File: python/src/xstudio/api/session/bookmark/annotation.py
# SPDX-License-Identifier: Apache-2.0
from xstudio.api.session.container import Container

class Annotation(Container):
    """Annotation object."""

    def __init__(self, connection, remote, uuid=None):
        """Create Annotation object.

        Args:
            connection(Connection): Connection object
            remote(actor): Remote actor object

        Kwargs:
            uuid(Uuid): Uuid of remote actor.
        """
        Container.__init__(self, connection, remote, uuid)

File: src/plugin/viewport_overlay/annotations/src/annotation.cpp
// SPDX-License-Identifier: Apache-2.0
#include <utility>

#include "annotation.hpp"
#include "annotations_tool.hpp"
#include "annotation_serialiser.hpp"

using namespace xstudio::ui::viewport;
using namespace xstudio;


Annotation::Annotation() : bookmark::AnnotationBase() {}

Annotation::Annotation(const utility::JsonStore &s) : bookmark::AnnotationBase() {

    AnnotationSerialiser::deserialise(this, s);
}

utility::JsonStore Annotation::serialise(utility::Uuid &plugin_uuid) const {

    plugin_uuid = AnnotationsTool::PLUGIN_UUID;
    return AnnotationSerialiser::serialise((const Annotation *)this);
}
